### ISSUE TITLE
Indexes for m:n relations

### ISSUE THREAD

**Issue by Benny-** [balderdashy/sails-postgresql#111]
The following sails models do not generate indexes for the many-to-many relationship.

```javascript
var Bar = module.exports = {

    attributes: {
        foos: {
            collection: 'foo',
            index: true,
            via: 'bars',
            dominant: true,
        },
    }
};
```

```javascript
var Foo = module.exports = {

    attributes: {
        bars: {
            collection: 'bar',
            index: true,
            via: 'foos',
        },
    }
};
```

The following table is generated in the postgresql database (bar and foo table not shown, they only contain a id):

```
Table name: bar_foos__foo_bars
  id: INT       <- indexed
  bar_foos: INT <- not indexed
  foo_bars: INT <- not indexed
```

Creating indexes for the foreign keys can speed up .populate() operations.

**Comment by tjwebb** [comment]
For now, you can manually set `index: true` on those fields. We don't create foreign key constraints either, although that might also be a task for improving associations in the future. We just need to make sure it wouldn't create incompatibilities with other adapters.

**Comment by sailsbot** [comment]
Thanks for posting, @Benny-.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
